mlflow/java/client/src/main/java/org/mlflow/tracking/ActiveRun.java

```diff
@@ -0,0 +1,95 @@
+package org.mlflow.tracking;
+
+import org.mlflow.api.proto.Service.*;
+
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * Represents an active MLflow run and contains APIs to log data to the run.
+ */
+public class ActiveRun {
+  private MlflowClient client;
+  private RunInfo runInfo;
+
+  public String getId() {
+    return runInfo.getRunId();
+  }
+
+  ActiveRun(RunInfo runInfo, MlflowClient client) {
+    this.runInfo = runInfo;
+    this.client = client;
+  }
+
+  public void logParam(String key, String value) {
+    client.logParam(getId(), key, value);
+  }
+
+  public void setTag(String key, String value) {
+    client.setTag(getId(), key, value);
+  }
+
+  public void logMetric(String key, double value) {
+    logMetric(key, value, 0);
+  }
+
+  public void logMetric(String key, double value, int step) {
+    client.logMetric(getId(), key, value, System.currentTimeMillis(), step);
+  }
+
+  public void logMetrics(Map<String, Double> metrics) {
+      logMetrics(metrics, 0);
+  }
+
+  public void logMetrics(Map<String, Double> metrics, int step) {
+    List<Metric> protoMetrics = metrics.entrySet().stream()
+      .map((metric) ->
+        Metric.newBuilder()
+          .setKey(metric.getKey())
+          .setValue(metric.getValue())
+          .setTimestamp(System.currentTimeMillis())
+          .setStep(step)
+          .build()
+      ).collect(Collectors.toList());
+    client.logBatch(getId(), protoMetrics, Collections.emptyList(), Collections.emptyList());
+  }
+
+  public void logParams(Map<String, String> params) {
+    List<Param> protoParams = params.entrySet().stream().map((param) ->
+      Param.newBuilder()
+        .setKey(param.getKey())
+        .setValue(param.getValue())
+        .build()
+    ).collect(Collectors.toList());
+    client.logBatch(getId(), Collections.emptyList(), protoParams, Collections.emptyList());
+  }
+
+  public void setTags(Map<String, String> tags) {
+    List<RunTag> protoTags = tags.entrySet().stream().map((tag) ->
+      RunTag.newBuilder().setKey(tag.getKey()).setValue(tag.getValue()).build()
+    ).collect(Collectors.toList());
+    client.logBatch(getId(), Collections.emptyList(), Collections.emptyList(), protoTags);
+  }
+
+  public void logArtifact(Path localPath) {
```

Comment:
We should probably generally document all these methods to a similar extent as `fluent.py`. We should expect that Java users may only read the documentation for this file (i.e., that they wouldn't go to `MlflowClient`'s docs.